src/main/java/com/compstore/service/impl/PCServiceImpl.java

```diff
@@ -141,6 +141,9 @@ private void fetchPCGraphicsCardBrand(PCEntity pcEntity, UUID graphicsCardBrandU
     }
 
     private void fetchPCOperatingSystem(PCEntity pcEntity, UUID operatingSystemUUID) {
+        if (operatingSystemUUID == null) {
```

Comment:
PC może być sprzedawany bez systemu, więc możliwe jest zapisanie go z nullem. Przez to jednak trzeba było zabezpieczyć wyszukiwanie systemu operacyjnego przed nullem